app/plugin/FeatureInfoWindow.js

```diff
@@ -0,0 +1,143 @@
+Ext.define('CpsiMapview.plugin.FeatureInfoWindow', {
+    extend: 'Ext.plugin.Abstract',
+    alias: 'plugin.cmv_feature_info_window',
+    pluginId: 'cmv_feature_info_window',
+
+    requires: [
+        'Ext.Component',
+        'CpsiMapview.view.window.MinimizableWindow',
+        'BasiGX.view.grid.FeaturePropertyGrid',
+        'BasiGX.util.Map'
+    ],
+
+    init: function () {
+        var me = this;
+
+        var mapComp = BasiGX.util.Map.getMapComponent();
+        var map = mapComp.getMap();
+
+        map.on('singleclick', function (evt) {
+            if (map.get('defaultClickEnabled')) {
+                me.requestFeatureInfos(mapComp, evt);
+            }
+        });
+    },
+
+    requestFeatureInfos: function (mapComp, evt) {
+        var me = this;
+
+        var map = mapComp.getMap();
+
+        var layers = [];
+        var resolution = map.getView().getResolution();
+        var projection = map.getView().getProjection();
+
+        var format = new ol.format.GeoJSON();
+
+        map.forEachLayerAtPixel(evt.pixel, function (layer) {
+            layers.push(layer);
+        }, undefined, function (layer) {
+            return layer.get('featureInfoWindow');
+        });
+
+        var win = me.openFeatureInfoWindow(map, evt);
+
+        if (layers.length <= 1) {
+            win.setLayout('fit');
+        } else {
+            win.setLayout({
+                type: 'accordion',
+                titleCollapse: false,
+                animate: true
+            });
+        }
+
+        if (layers.length === 0) {
+            win.add(Ext.create('Ext.Component', {
+                html: '<span>No results found</span>'
+            }));
+        } else {
+            layers.forEach(function (layer) {
+                var url = layer.getSource().getGetFeatureInfoUrl(evt.coordinate, resolution, projection, {
+                    INFO_FORMAT: 'geojson'
+                });
+
+                Ext.Ajax.request({
+                    url: url
+                }).then(function (response) {
+                    var features = format.readFeatures(response.responseText);
+                    win.add(me.createFeaturePanels(mapComp, layer, features));
+                }).then(undefined, function (error) {
+                    Ext.log(error);
+                    console.error(error);
+                });
+            });
+        }
+    },
+
+    openFeatureInfoWindow: function (map, evt) {
+        var me = this;
+
+        if (!this.highlightSource) {
+            this.highlightSource = new ol.source.Vector();
```

Comment:
Is there a special reason why `highlightSource` is a member var since it is just used in `openFeatureInfoWindow`? If yes, please declare and comment it around line 12.

Reply:
It is stored so it is not recreated every time.